table/tables/tables.go

```diff
@@ -323,6 +323,20 @@ func (t *Table) rebuildIndices(ctx context.Context, h int64, touched []bool, old
 	return nil
 }
 
+func (t *Table) delDirtyIndexKey(createdIdx int, recordID int64, r []interface{}, txn kv.Transaction) {
+	for i, v := range t.indices {
+		if v == nil {
+			continue
+		}
+
+		if i >= createdIdx {
+			break
+		}
+		colVals, _ := v.FetchValues(r)
+		_ = v.X.Delete(txn, colVals, recordID)
```

Comment:
@winkyao  you can supply a CheckContraint function or other, so that before insert and update, we can use this to check whether we can handle this row or not. :-)